libfreerdp/crypto/certificate.c

```diff
@@ -1172,7 +1180,15 @@ void certificate_free_int(rdpCertificate* cert)
 	if (cert->x509)
 		X509_free(cert->x509);
 	if (cert->chain)
+	{
+		const int count = sk_X509_num(cert->chain);
```

Comment:
All of the code inside the if could be replaced by `sk_X509_pop_free(cert->chain, x509_free);`